extension/js/common/browser/browser-msg.ts

```diff
@@ -449,6 +454,40 @@ export class BrowserMsg {
     });
   }
 
+  public static thunderbirdSecureComposeHandler() {
+    const handleClickEvent = async (tabId: number, acctEmail: string, thunderbirdMsgId: number, composeMethod?: browser.compose._ComposeDetailsType) => {
+      const accountEmails = await GlobalStore.acctEmailsGet();
+      const useFullScreenSecureCompose = (await browser.windows.getCurrent()).type === 'messageCompose';
+      composeMethod = composeMethod === 'reply' || composeMethod === 'forward' ? composeMethod : undefined;
+      if (accountEmails.length !== 0) {
+        await BgUtils.openExtensionTab(
+          Url.create('/chrome/settings/inbox/inbox.htm', { acctEmail, useFullScreenSecureCompose, thunderbirdMsgId, composeMethod })
+        );
+        await browser.tabs.remove(tabId);
+      } else {
+        await BgUtils.openExtensionTab(Url.create('/chrome/settings/initial.htm', {}));
+      }
+    };
+    browser.composeAction.onClicked.addListener(async tab => {
```

Comment:
I also noticed that it's recommended to use `messenger...` instead of `browser...` in thunderbird environment - https://webextension-api.thunderbird.net/en/128-esr-mv3/#getting-started, I think it'll be useful for differentiating browser and thunderbird actions

Reply:
Thanks @sosnovsky - I have settled this one.